vertx-grpcio-common/src/main/java/io/vertx/grpcio/common/impl/Utils.java

```diff
@@ -57,7 +57,7 @@ public static Metadata readMetadata(MultiMap headers) {
     byte[][] array = new byte[entries.size() * 2][];
     int idx = 0;
     for (Map.Entry<String, String> entry : entries) {
-      String key = entry.getKey();
+      String key = entry.getKey().toLowerCase();
```

Comment:
`toLowerCase()` is locale sensitive, this should be:

```java
      String key = entry.getKey().toLowerCase(Locale.ROOT);
```

Reply:
makes sense. fixed!